lib/footer.js

```diff
@@ -108,8 +108,8 @@ export function footer(h) {
     children: [
       {
         type: 'element',
-        tagName: 'h2',
-        properties: {id: 'footnote-label', className: ['sr-only']},
+        tagName: h.footnoteLabelTagName,
+        properties: {id: 'footnote-label', ...h.footnoteLabelProperties},
```

Comment:
`...` makes a shallow clone.
Because props can be an object that contains arrays, those arrays are not deeply cloned, but are instead the same array.
To solve this, create a deep clone, by passing `h.footnoteLabelProperties` through `JSON.parse(JSON.stringify(x))`.

Additionally, why is `id` not in the defaults for this? Why should it be forced, always?

Reply:
Thanks a lot! I am definitely learning something here.

For the id, i think i had written it in this way so that if the user does not input any id in the configuration, there will still be a 'footnote-label' id generated by default. If there is an id in the user-provided object, the spread operator will overwrite the existing one so the id is still configurable. In order to have no id, users could set id: null.

I think if i change index.js to have the defaults there instead, users will have to explicitly write an id, even if they only want to change the class.

I'm going to push an updated patch with both id and className in the defaults, let me know what approach you prefer!

I will also add documentation to explain the options so you can review this as well.